cylc/flow/network/schema.py

```diff
@@ -857,340 +889,689 @@ class Meta:
         return GenericResponse(result="Error: No given Workflow(s)")
     if exworkflows is None:
         exworkflows = []
-    w_args = {}
-    w_args['workflows'] = [parse_workflow_id(w_id) for w_id in workflows]
-    w_args['exworkflows'] = [parse_workflow_id(w_id) for w_id in exworkflows]
+    # w_args = {}
+    # w_args['workflows'] = [parse_workflow_id(w_id) for w_id in workflows]
+    # w_args['exworkflows'] = [parse_workflow_id(w_id) for w_id in exworkflows]
+    w_args = {
+        'workflows': workflows,
+        'exworkflows': exworkflows
+    }
     if args.get('args', False):
         args.update(args.get('args', {}))
         args.pop('args')
     resolvers = info.context.get('resolvers')
     res = await resolvers.nodes_mutator(info, command, ids, w_args, args)
     return GenericResponse(result=res)
 
+# Input types:
+
+
+class WorkflowName(String):
+    """The name of the registered workflow."""
+
+
+class User(String):
+    """The user the workflow is registered under."""
+
+
+class WorkflowID(InputObjectType):
+    """A registered workflow."""
+
+    name = WorkflowName(required=True)
+    # NOTE: at present we only support single-user queries
+    #       if the user field is blank assume the workflow owner
+    user = User()
```

Comment:
Changed from a string `<user>|<name>:<status>` to an object `{user, name, status}`, good or bad?

Reply:
Neither but simpler to go down the string route for now.